### PR TITLE
Pyproject toml

### PR DESCRIPTION
**Pull request by kyleaoman**
Move from setup.py to pyproject.toml. Notes:

- The version is hardcoded in the pyproject.toml file instead of read from the __version__ file. Some googling suggests that there is not way around this that doesn't make more of a mess than it's worth.
- I didn't include a long description, but pyproject.toml lets you point to the readme file, so I think that covers the same role.

### CODE REVIEW

**Comment by JBorrow** [comment]
As discussed before, I think simply adding the build environment pyproject.toml should suffice here :)

**Comment by kyleaoman** [comment]
@JBorrow I put back the setup.py file.

**Comment by MatthieuSchaller** [comment]
@JBorrow any last thoughts?

**Comment by JBorrow** [comment]
Can pyproject.toml not be much reduced now we are still using the old build system? I think all you need to do is specify that it uses setuptools.

**Comment by MatthieuSchaller** [comment]
@kyleaoman any thoughts on Josh's last point?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+[tool.setuptools]
+packages = ["velociraptor"]
+
+[project]
+name = "velociraptor-python"
+version="0.16.1"
+authors = [
+    { name="Josh Borrow", email="joshua.borrow@durham.ac.uk" },
+]
```
Comment (JBorrow) [inline pyproject.toml:13]:
```suggestion
authors = [
    { name="Josh Borrow", email="josh@joshborrow.com" },
    { name="Kyle Oman", email="kyle.a.oman@durham.ac.uk" },
]
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+description="Velociraptor catalogue reading routines."
+readme = "README.md"
+requires-python = ">3.6.0"
+classifiers = [
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
```
Comment (JBorrow) [inline pyproject.toml:20]:
```suggestion
```

**Review by JBorrow — COMMENTED** [review]
Ok, I am happy with this now.

**Comment by JBorrow** [comment]
I think we should merge this and then re-visit changing from setup.py to pyproject.toml from there. Kyle, we should also get you write access to the repo.

**Comment by MatthieuSchaller** [comment]
Kyle is already a maintainer. 

Note that this whole project should be destined for death pretty soon. We should access the SOAP catalogs directly via swiftsimio without the need of an intermediate tool.

**Comment by MatthieuSchaller** [comment]
Indeed, but that should really disappear too.

**Comment by JBorrow** [comment]
As in it will be re-written, or moved?

**Comment by MatthieuSchaller** [comment]
neither at this stage.

**Comment by kyleaoman** [comment]
When this is killed off I'd appreciate a bit of warning so that I can update swiftgalaxy, which relies heavily on this for docs and tests. Would like to move that to SOAP first, but that implies having a functional i/o package for SOAP.

**Comment by kyleaoman** [comment]
Should I deal with the suggested changes and merge this in, then? Probably not worth the bother woth pyproject.toml if this is getting mothballed anyway.

**Comment by JBorrow** [comment]
I think that makes the most sense, thanks Kyle.

**Comment by kyleaoman** [comment]
@JBorrow I think that I've done all that I can on my end, merging is blocked until there's an approving review (and I can't approve as the PR author).

**Pull request merged**